unity/CAVE/Assets/W3D_Translator/CLI.cs

```
@@ -19,106 +21,134 @@ static void Main()
+        W3D.Camera xmlCaveCamera = xml.CaveCamera;
+        UnityEngine.Camera caveCamera = 
+            caveCameraT.GetComponent<UnityEngine.Camera>();
+        caveCamera.farClipPlane = xmlCaveCamera.FarClip;
+        caveCamera.transform.localPosition = 
+            Xml.ConvertVector3(xmlCaveCamera.Placement.PositionString);
+
+        // Update Camera inside of xrRig
+        W3D.Camera xmlCamera = xml.Camera;
+        UnityEngine.Camera camera = mainCameraT.GetComponent<UnityEngine.Camera>();
+        camera.farClipPlane = xmlCamera.FarClip;
+        xrRig.transform.position = 
+            // xml.Camera is really the player's position - update xrRig directly
+            // xrRig is outside the Story object so we must convert to meters
+            Xml.ConvertVector3(xmlCamera.Placement.PositionString) * 0.3048f;
+
+        // Update tracking settings for the Main Camera
+        TrackedPoseDriver tracking = mainCameraT.GetComponent<TrackedPoseDriver>();
+        bool allowRotation = xml.WandNavigation.AllowRotation;
+        bool allowMovement = xml.WandNavigation.AllowMovement;
+        Debug.Log($"xml: {xmlCamera} {allowRotation} {allowMovement}");
+        if(!allowRotation && !allowMovement) {
+            tracking.enabled = false;
+            // Setting the tracking to device based re-adds the camera offset
+            xrRig.GetComponent<XROrigin>().RequestedTrackingOriginMode = 
+                XROrigin.TrackingOriginMode.Device;
+            Debug.Log(tracking.ToString());
+        }
+        else {
+            # pragma warning disable CS8509 // (false, false) case is handled above
+            tracking.trackingType = (allowRotation, allowMovement) switch {
+                (true, true) => TrackedPoseDriver.TrackingType.RotationAndPosition,
+                (true, false) => TrackedPoseDriver.TrackingType.RotationOnly,
+                (false, true) => TrackedPoseDriver.TrackingType.PositionOnly,
+            };
+        }
```

This is a little difficult for me to parse. Maybe something like:

```
switch (allowRotation, allowMovement) {
  case (true, true):
    tracking.trackingType = TrackedPoseDriver.TrackingType.RotationAndPosition
    break;
  case (true, false):
    tracking.trackingType = TrackedPoseDriver.TrackingType.RotationOnly;
    break;
  case (false, true):
    tracking.trackingType = TrackedPoseDriver.TrackingType.RotationOnly;
    break;
  default:
                tracking.enabled = false;
            // Setting the tracking to device based re-adds the camera offset
            xrRig.GetComponent<XROrigin>().RequestedTrackingOriginMode = 
                XROrigin.TrackingOriginMode.Device;
            Debug.Log(tracking.ToString());
    break;
}
```

---

Ah yes, I got so excited about the pattern matching it appears I've thrown out any code readability! I'll use `(false, false)` instead of a default case